pkg/controller.v2/jobcontroller/jobcontroller.go

```diff
@@ -115,6 +133,75 @@ type JobController struct {
 	Recorder record.EventRecorder
 }
 
+func NewJobController(
```

Comment:
The name may conflicts users since Kubernetes has a job controller

Reply:
How about TrainingJobController or BaseController

Reply:
Since it is package scoped, is it a problem?  I initially thought of putting just "Controller" as the package name.  But I felt that it will be confusing as we already have a "Controller" package for v1alpha1.

Reply:
The new contributors will be confused, maybe. While it is not a big problem, just a nit.